ophyd/signal.py

```diff
@@ -780,15 +793,16 @@ def put(self, value, force=False, **kwargs):
         if not force:
             self.check_value(value)
 
-        if not self._write_pv.connected:
-            if not self._write_pv.wait_for_connection():
-                raise TimeoutError('Failed to connect to %s' %
-                                   self._write_pv.pvname)
-
-        use_complete = kwargs.pop('use_complete', self._put_complete)
+        with self._lock:
+            if not self._write_pv.connected:
+                if not self._write_pv.wait_for_connection():
+                    raise TimeoutError('Failed to connect to %s' %
+                                       self._write_pv.pvname)
 
-        self._write_pv.put(value, use_complete=use_complete,
-                           **kwargs)
+            use_complete = kwargs.pop('use_complete', self._put_complete)
```

Comment:
Off-topic for this PR, but as long as you're touching this, would you mind making it a proper kwarg for readability's sake?